src/ticks.js

```diff
@@ -75,8 +75,8 @@ export function validTicks(scale, ticks, count) {
  * @param {*} [count] - The approximate number of desired ticks.
  * @return {Array<*>} - The generated tick values.
  */
-export function tickValues(scale, count) {
-  return scale.ticks ? scale.ticks(count) : scale.domain();
+export function tickValues(scale, count, integer) {
+  return scale.ticks ? scale.ticks((integer == null || !integer)?count:((scale.domain()[1] < count)?scale.domain()[1]:count)) : scale.domain();
```

Comment:
This code is error prone. For example scale domains can have more than two elements. There is no guarantee that `domain[1]` is the maximum value. FWIW, I tend to use `peek(domain)` to pull last the array value.